api/topo/topo.go

```diff
@@ -33,3 +58,86 @@ var TopoClientFactory = func(cc *grpc.ClientConn) TopoClient {
 func CreateTopoClient(cc *grpc.ClientConn) TopoClient {
 	return TopoClientFactory(cc)
 }
+
+// ObjectToDevice ...
+func ObjectToDevice(obj *Object) *device.Device {
+	if obj == nil || obj.Type != Object_ENTITY { // Device is an entity
+		return nil
+	}
+
+	d := &device.Device{
+		ID:        device.ID(obj.ID),
+		Protocols: []*device.ProtocolState{},
+	}
+
+	r, _ := strconv.Atoi(obj.Attributes[Revision])
+	d.Revision = device.Revision(r)
+	d.Address = obj.Attributes[Address]
+	d.Target = obj.Attributes[Target]
```

Comment:
I think you should throw error if `address`, `type` or `version` are not found.

Will it be getting `type` from the Entity `kind`?

Reply:
done